Control/Parallel/Strategies.hs

```diff
@@ -296,13 +296,26 @@ withStrategy = flip using
 -- | Compose two strategies sequentially.
 -- This is the analogue to function composition on strategies.
 --
--- For any strategies @strat1@, @strat2@, and @strat3@,
+-- > strat2 `dot` strat1 == strat2 . withStrategy strat1
+--
+-- 'dot' is associative:
 --
 -- > (strat1 `dot` strat2) `dot` strat3 == strat1 `dot` (strat2 `dot` strat3)
--- > strat1 `dot` strat1 = strat1
--- > strat1 `dot` r0 == strat1
 --
--- > strat2 `dot` strat1 == strat2 . withStrategy strat1
+-- 'r0', 'rseq', and 'rpar' are all one-sided identities of 'dot':
+--
+-- > strat `dot` r0 == strat
+-- > strat `dot` rseq == strat
+-- > strat `dot` rpar == strat
```

Comment:
Is that true?

Reply:
I believe so, yes. `runEval m` is a lifted value; it doesn't do anything until it's forced. Forcing `runEval (r0 x)` runs the usual `return x` `IO` computation, so that gives you `x`. Forcing `runEval (rseq x)` runs an `IO` computation that forces `x` to WHNF and returns the result (exactly the same as forcing `x` directly). Forcing `runEval (rseq x)` runs an `IO` computation that sparks a thread to evaluate `x` to WHNF, but immediately forces the result so the thread (almost certainly) fizzles.

Reply:
```Haskell
r0 `dot` rseq == r0
r0 `dot` rpar == r0
```
Would be counterintuitive to my expectations, of the first description of `dot`, namely: "Compose two strategies sequentially". Or more explicitely:

"Composing the 'do nothing'-strategy with the 'evaluate to WHNF'-strategy sequentially, creates the 'do nothing'-strategy"

Am I the only, who thinks this sounds weird?

Reply:
``r0 `dot` rseq`` is *not* `r0`. But ``rseq `dot` r0 = r0``.

Reply:
Last comment corrected.

Reply:
`dot` is a composition operator in a `semigroupoids` sense: namely, it is associative. But it doesn't form a category, which makes intuition around it a little tricky. @Borgvall, by definition,

```haskell
f `dot` g = \x -> f (runEval (g x))
```

Think about what `runEval (r0 x)`, `runEval (rpar x)`, and `runEval (rseq x)` do when forced. Ultimately the same thing (if you ignore sparks that fizzle). If you don't force them, they all do ... nothing.

Reply:
I care about sparking, this is the main reason to use this library!

Reply:
I'm still concerned about `strat `dot` rpar == strat`

We can construct contrived examples in which this doesn't apply, at least operationally (and operationally is after all the point of the library, so the equalities should be operational ones). e.g.

```
let strat n x = rseq (fib n) >> return x  in strat 99999 `dot` rpar
```

If `strat` is very slow then the spark doesn't fizzle immediately and we get some useful parallelism. Of course it's hard to reliably define a strategy that's very slow (the example above could easily be optimised), but the fact that it's possible at all implies that we can't assume the spark fizzles immediately.